modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/renewals-chart/index.js

```diff
@@ -10,75 +11,110 @@
  */
 
 import ClayButton from '@clayui/button';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
 import ClassNames from 'classnames';
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 
 import Container from '../../common/components/container';
 
-const items = [
-	{
-		name: 'Rent-A-Centre Texas, L.P',
-		status: 30,
-	},
-	{
-		name: 'Advance Stores Co, INC',
-		status: 60,
-	},
-	{
-		name: 'Mission of Hope Haiti',
-		status: 60,
-	},
-	{
-		name: 'Rite Aid',
-		status: 90,
-	},
-];
-
 const status = {
-	30: 'bg-accent-1',
-	60: 'bg-warning',
-	90: 'bg-success',
+	5: 'bg-accent-1',
+	15: 'bg-warning',
+	30: 'bg-success',
 };
 
 export default function () {
+	const [data, setData] = useState();
+	const currentDate = new Date();
+
+	useEffect(() => {
+		const getRenewalsData = async () => {
+			// eslint-disable-next-line @liferay/portal/no-global-fetch
+			await fetch('/o/c/opportunitysfs?&sort=closeDate:asc', {
+				headers: {
+					'accept': 'application/json',
+					'x-csrf-token': Liferay.authToken,
+				},
+			})
+				.then((response) => response.json())
+				.then((data) => {
+					setData(data);
+				})
+				.catch((error) => {
+					console.error('Error:', error);
```

Comment:
acredito que não usamos o console pro catch error. Tu pode usar
```suggestion
					Liferay.Util.openToast({
		message: 'An unexpected error occured.',
		type: 'danger',
	});
```